nbd_http_server.py

```diff
@@ -274,10 +280,17 @@ def clean_paths():
 
     # Continue to log server messages in stdout.
     def log_server_messages():
+        verbose_keywords = [
+            'pread',
+            'pwrite'
+        ]
         while server.poll() is None:
             line = server.stdout.readline().rstrip('\n')
             if line:
-                thread_print(line)
+                if any(keyword in line for keyword in verbose_keywords):
+                    dprint(line)
+                else:
+                    thread_print(line)
```

Comment:
It think that we can use `any(a.startswith(keyword) for keyword in verbose_keywords)` instead.

Reply:
There is a bunch of elements in front of the actual message, so I'd guess that `in` instead of `startswith` is relevant.

One of the logs being `nbdkit: multihttp[1]: debug: pread count=131072 offset=0`

Reply:
I don't like this log prefix. :'(
But if we don't have the choice (no option to remove it), we can keep your change.

Reply:
I have not found any flag / option in nbdkit to filter / adjust that. I can do something like this if it's ok, all debug logs seems to have this constant prefix.

```python
NBDKIT_LOG_PREFIX = "nbdkit: multihttp[1]: debug: "
...
def log_server_messages():
    verbose_keywords = (
        'pread',
        'pwrite'
    )
    while server.poll() is None:
        line = server.stdout.readline().rstrip('\n')
        if line:
            if any(line.startswith(NBDKIT_LOG_PREFIX + keyword) for keyword in verbose_keywords):
                dprint(line)
            else:
                thread_print(line)
```

(Assuming the global constant would be placed somewhere more relevant ofc)

Reply:
with something like this:
```python
    def log_server_messages():
        nbdkit_log_prefix = r'nbdkit: multihttp\[\d+\]: debug: '
        verbose_keywords = (
            'pread',
            'pwrite'
        )
        verbose_pattern = re.compile(r'{}({})'.format(nbdkit_log_prefix, '|'.join(verbose_keywords)))

        while server.poll() is None:
            line = server.stdout.readline().rstrip('\n')
            if line:
                if any(verbose_pattern.match(line) for keyword in verbose_keywords):
                    dprint(line)
                else:
                    thread_print(line)
```

Reply:
added my last suggestion in [d8244e8](https://github.com/xcp-ng/http-nbd-transfer/pull/14/commits/d8244e83819e45876c5b76c778b8e8fd73ab403e)